### PR TITLE
refac(core): use only Discord.js Client

### PR DESCRIPTION
**Pull request by joaoscheuermann**

### CODE REVIEW

File: src/oxr/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  readonly [key: string]: string | number | boolean;
+}
+
+export interface OXEDollarRates {
+  rates: Record<string, number>;
+  base: string;
+  timestamp: number;
+  license: string;
+  disclaimer: string;
+}
```
Comment (CaioCS97) [inline src/oxr/index.ts:16]: OXE or OXR? 👀

File: src/oxr/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+const request = (method: 'GET', path: string, query: OXEQueryParams): Promise<Response> => {
+  const url = `https://openexchangerates.org/api` + path + Object.entries(query).reduce((str, [key, value]) => str + `${str.length ? '&' : '?'}${key}=${value}`, '')
```
Comment (CaioCS97) [inline src/oxr/index.ts:19]: I would move this reduce to a getQuery function:
```suggestion
  const getQuery = (query: OXEQueryParams) => Object.entries(query).reduce((str, [key, value]) => str + `${str.length ? '&' : '?'}${key}=${value}`, '');
  // [...]
  const url = `https://openexchangerates.org/api${path}${getQuery(query)}`
```
WDYT?

File: src/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { CommandInteraction, SlashCommandBuilder } from "discord.js";
+
+export interface DolinhoCommand {
+  data: SlashCommandBuilder,
+  execute: (interaction: CommandInteraction) => Promise<void>,
+}
```
Comment (CaioCS97) [inline src/types/index.ts:6]: missing empty line

**Review by CaioCS97 — CHANGES_REQUESTED** [review]

File: src/oxr/index.ts (hunk 1 above)
Comment (CaioCS97) [reply src/oxr/index.ts:16]: also. why not move this types to the types file?

File: src/oxr/index.ts (hunk 2 above)
Comment (joaoscheuermann) [reply src/oxr/index.ts:19]: I think this doesn't make much of a difference...

File: src/oxr/index.ts (hunk 1 above)
Comment (joaoscheuermann) [reply src/oxr/index.ts:16]: good call, will fix for OXR

File: src/oxr/index.ts (hunk 2 above)
Comment (CaioCS97) [reply src/oxr/index.ts:19]: It's just to improve readability.
Also, we can reuse the the getQuery function to build params string on further implementations.
That said, I don't really have problems with the way it is right now.

**Review by CaioCS97 — APPROVED** [review]

**Pull request merged**